libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/router/RouterInterfaceAdapter.kt

```diff
@@ -124,12 +123,39 @@ class RouterInterfaceAdapter(
 
     override fun getRouteRefresh(
         options: RouteRefreshOptions,
-        route: String,
+        callback: RouterRefreshCallback
+    ): Long {
+        val route = getCurrentRoutesFun().find {
+            it.directionsResponse.uuid() == options.requestId
+                && it.routeIndex == options.routeIndex
+        }
+        if (route != null) {
+            return getRouteRefresh(options, route, callback)
+        } else {
+            callback.run(
+                ExpectedFactory.createError(
+                    RouterError(
+                        "There is no route that matches $options, nothing to refresh.",
+                        ROUTE_REFRESH_FAILED_DEFAULT_CODE,
+                        RouterErrorType.UNKNOWN,
+                        -1
+                    )
+
+                ),
+                com.mapbox.navigator.RouterOrigin.ONLINE
```

Comment:
should be custom for user-provided interfaces
```suggestion
                com.mapbox.navigator.RouterOrigin.CUSTOM
```